### PR TITLE
Feature/building occlusion

### PR DESCRIPTION
**Pull request by wbeltran777**
Sphere occlusion, now you can see behind building. (In the test scene for now, next pull request will likely be about implementing everything)

CullSphere is part of the player prefab now. 
Camera Culling script was changed to work with the new culling method. 

yellow.mat is a material I used for testing out the occlusion in the test scene.

### CODE REVIEW

File: Assets/_PartyParkade/Script/Game/CameraCulling.cs [hunk 1]
```diff
@@ -30,15 +40,21 @@ private void OnDestroy()
...
-                Mathf.Infinity)) return;
-            //// If anything that it hits has an ObjectHiding script, do soemthing about it.
-            ObjectHiding obj = hit.collider.gameObject.GetComponent<ObjectHiding>();
-            if (obj == null) return;
-            obj.Hide();
-            obj.SetSurroundingTransparent();
+
+            if (Physics.Raycast(this.transform.position,
+                (m_cullSphere.transform.position - this.transform.position).normalized,
+                out hit, Mathf.Infinity))
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/CameraCulling.cs:46]: i don't think u need Math.infinity, i think that's defaulted

Comment (wbeltran777) [reply Assets/_PartyParkade/Script/Game/CameraCulling.cs:46]: You're super right, will change right away

File: Assets/_PartyParkade/Script/Game/CameraCulling.cs [hunk 2]
```diff
@@ -30,15 +40,21 @@ private void OnDestroy()
...
+            {
+                // Increase size of cull sphere if car is behind an "Obstacle" 
+                if (!hit.collider.gameObject.CompareTag(Data.Tags.Obstacles))
+                {
+                    m_cullSphere.transform.DOScale(new Vector3(0, 0, 0), 0.5f);
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/CameraCulling.cs:51]: Put the 0.5 as a const

File: Assets/_PartyParkade/Script/Game/CameraCulling.cs [hunk 3]
```diff
@@ -30,15 +40,21 @@ private void OnDestroy()
...
+                out hit))
+            {
+                // Increase size of cull sphere if car is behind an "Obstacle" 
+                if (!hit.collider.gameObject.CompareTag(Data.Tags.Obstacles))
+                {
+                    m_cullSphere.transform.DOScale(new Vector3(0, 0, 0), 0.5f);
+                }
+                else
+                {
+                    m_cullSphere.transform.DOScale(m_originalSphereScale, 0.5f);
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/CameraCulling.cs:55]: val scale = Vector3.zero;
if (condition) {
scale = spherescale
}

m_cullSphere.transform.DOScale(scale , someConst);

Comment (wbeltran777) [reply Assets/_PartyParkade/Script/Game/CameraCulling.cs]: How's it look now?

**Comment by m4rtint** [comment]
Looks good

**Comment by DJSteins** [comment]
@wbeltran777 if you wrote the shader files then can you add some comments to explain what is happening in it?

**Comment by wbeltran777** [comment]
> @wbeltran777 if you wrote the shader files then can you add some comments to explain what is happening in it?

Sure! I'll push some new changes soon to talk about the important bits :)

**Comment by mikeslgoh** [comment]
@DJSteins can you help to review the PR again and see if that helps?

File: Assets/_PartyParkade/Shaders/CullSphereShader.shader [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			// This is the "color" function mentioned earlier
+            fixed4 frag (v2f i) : SV_Target
+            {
+                // col is a vector4, and we take a sample from the texture
+				// tex2D gets the main texture and uv values 
+                fixed4 col = tex2D(_MainTex, i.uv);
+				// col.a is the color's 'alpha' 
+				// We set it to our desired tansparency (in this case 1, which we set from the inspector)
+				col.a = _Transparency;
+
```
Comment (DJSteins) [inline Assets/_PartyParkade/Shaders/CullSphereShader.shader:92]: Can you check the indentations in this file? Some lines look over-indented.

**Comment by mikeslgoh** [comment]
[Test 3D building culling](https://trello.com/c/E3O47L2u/383-test-3d-building-culling)

File: Assets/_PartyParkade/Shaders/CullSphereShader.shader (hunk 4 above)
Comment (DJSteins) [reply Assets/_PartyParkade/Shaders/CullSphereShader.shader:92]: @wbeltran777 Please check all of the indentations again as they show on GitHub. They still aren't aligned.

File: Assets/_PartyParkade/Shaders/ObstacleShader.shader [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        half _Glossiness;
+        half _Metallic;
+        fixed4 _Color;
+
+        // Add instancing support for this shader. You need to check 'Enable Instancing' on materials that use the shader.
+        // See https://docs.unity3d.com/Manual/GPUInstancing.html for more information about instancing.
+        // #pragma instancing_options assumeuniformscaling
+        UNITY_INSTANCING_BUFFER_START(Props)
+            // put more per-instance properties here
+        UNITY_INSTANCING_BUFFER_END(Props)
```
Comment (DJSteins) [inline Assets/_PartyParkade/Shaders/ObstacleShader.shader:65]: Indentations in this file too need to be fixed.

**Pull request merged**